tests/ignite/metrics/test_recall.py

```diff
@@ -540,7 +544,8 @@ def update(engine, i):
     metric_devices = ["cpu"]
     if device.type != "xla":
         metric_devices.append(idist.device())
-    for _ in range(2):
+    for i in range(2):
+        torch.manual_seed(12 + rank + i)
```

Comment:
So you could remove line 497 and bring line 496 near here.

Reply:
Good catch @sadra-barikbin! 

@puhuk please ensure that you do not leave `torch.manual_seed(12)` in other metric tests